code/script.js

```diff
@@ -26,22 +34,130 @@ const showMessage = (message, sender) => {
+  select.addEventListener('change', () => handleContinentQuestion(select.value))
+}
+
+const weatherQuestion = () => {
+  showMessage('What kind of weather do you prefer?', 'bot')
+  inputWrapper.innerHTML = /*html*/ `
+  <button id="sun" value="sun">Sunny</button>
+  <button id="cloud" value="cloud">Cloudy</button>
+  <button id="snow" value="snow">Snowy</button>
+  `
+  document
+  .getElementById('sun')
+  .addEventListener('click', () => handleWeatherQuestion('sunny'))
+  document
+  .getElementById('cloud')
+  .addEventListener('click', () => handleWeatherQuestion('cloudy'))
+  document
+  .getElementById('snow')
+  .addEventListener('click', () => handleWeatherQuestion('snowy'))
+}
+
+
+const bookQuestion = () => {
+  showMessage(`Perfect! We have found a destination in ${continent} where the weather is always as you want. Would you like to book a ticket?`, 'bot')
+  inputWrapper.innerHTML = /*html*/ `
+  <button id="yes" value="yes">Hell yeah!</button>
+  <button id="no" value="no">No, thanks.</button>
+  `
+  document
+  .getElementById('yes')
+  .addEventListener('click', () => handleBookQuestion('Hell yeah!'))
+  document
+  .getElementById('no')
+  .addEventListener('click', () => handleBookQuestion('No, thanks.'))
+}
+
+const lastMessage = () => {
+    showMessage(`Thank you, bye!`, 'bot')
+    inputWrapper.innerHTML = ``
+}
+
+// const lastMessage = () => {
+//   if (book === 'yes'){
+//     showMessage(`Thank you ${yourName}! Your booking has been sent tou you`, 'bot')
+//     inputWrapper.innerHTML = ''
+//   } else {
+//     showMessage(`Sorry to see you go, ${yourName}`, 'bot')
+//     inputWrapper.innerHTML = ''
+//   }
+// }
+
+const handleInput = (event1) => {
+  event1.preventDefault()
+  currentQuestion++
+  // console.log('our currentQuestion variable is:', currentQuestion)
+  if (currentQuestion === 1) {
+    handleNameQuestion()
+  } else if (currentQuestion === 2) {
+    handleContinentQuestion()
+  } else if (currentQuestion === 3) {
+    handleWeatherQuestion() 
+  } else if (currentQuestion === 3) {
+    handleBookQuestion() 
+  } 
+}
+
+const handleNameQuestion = () => {
+  yourName = inputValue.value
+  // console.log('the name is:', yourName)
+  showMessage(`My name is ${yourName}`, 'user' )
+  inputValue.value = ''
+  setTimeout(continentQuestion, 1000)
+}
+
+const handleContinentQuestion = () => {
+  continent = select.value
+  // console.log('the continent is:', continent)
+  showMessage(`I would like to visit ${continent}`, 'user' )
+  inputValue.value = ''
+  setTimeout(weatherQuestion, 1000)
 }
 
+const handleWeatherQuestion = (weather) => {
+  // console.log(`The weather is`, weather)
+  showMessage(`I'll like the weather to be ${weather}`, 'user' )
+  inputValue.value = ``
+  setTimeout(bookQuestion, 1000)
+}
+
+const handleBookQuestion = (book) => {
+  // console.log(`The booking is`, book)
+  showMessage(book, 'user' )
+  inputValue.value = ``
+  setTimeout(lastMessage, 1000)
+}
+
```

Comment:
I like how you split up the functionality of the asking each question versus the functionality of handling each response. It makes it very easy to follow what each function's purpose is.